production/helm/loki/templates/service-headless.yaml

```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "loki.name" . }}-headless
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "loki.labels" . | nindent 4 }}
+    variant: headless
+spec:
+  clusterIP: None
+  ports:
+    - name: http-metrics
+      port: 3100
+      targetPort: http-metrics
+      protocol: TCP
+  selector:
+    {{- if eq (include "loki.requiresMultiStageUpgrade" . ) "true" }}
+    {{- include "loki.legacy.singleBinarySelectorLabels" . | nindent 4 }}
+    {{- else -}}
+    {{- include "loki.selectorLabels" . | nindent 4 }}
+    {{- end }}
```

Comment:
Why is this required?

Reply:
That's a good question and I've decided it's not, see my refactor.

I do think we'll need something like this when we figure out how to do a zero downtime migration, but I only think we need that for people migrating from distributed or enterprise SSD charts. Since this PR only address the migration from filesystem backed single binary, I think it's safe to assume HA is not required.

The idea here is that if we want to be able to spin up the new cluster along-side an existing cluster to do a multi-stage, zero downtime deployment, we'll need the new deployment to include the legacy label set, and we'll need to new services to use the legacy label selector so it captures both new and old pods. However, like I mentioned, not needed yet.

Reply:
Thanks for explaining!